### PR TITLE
refactor to save n restore previous theme

### PR DESCRIPTION
**Pull request by edouard-lopez**
@jorgebucaran I refactored ayu-theme:

* saving color on install
* restoring on uninstall
* enabling via a universal variable `ayu_variant`
* updated the doc

What do you think?

### CODE REVIEW

**Comment by edouard-lopez** [comment]
## Installation

For the PR you fetch the code, checkout the branch and install the using the directory. 

    git clone https://github.com/edouard-lopez/ayu-theme.fish.git
    git checkout refactor-to-save-n-restore-previous-theme
    cd ayu-theme.fish/
    fisher install ./

## Configuration

Declare the variant you want as a universal variable, _e.g._:

```fish
set --universal ayu_variant light && ayu_load_theme
```

File: README.md [hunk 1]
```diff
@@ -8,9 +8,10 @@
 
 ## Configuration
 
-Edit `~/.config/fish/config.fish` to load a variant, _e.g._:
+Declare the variant you want as a universal variable, _e.g._:
 
-    source $__fish_config_dir/conf.d/ayu-light.fish
+    set --universal ayu_variant light
+    exec fish
```
Comment (jorgebucaran) [inline README.md]: It would be nicer if didn't have to `exec`-away our session to apply changes.

**Comment by edouard-lopez** [comment]
@jorgebucaran I wrapped the `switch` in a `ayu_load_theme` so we can just 

    set --universal ayu_variant light && ayu_load_theme

**Comment by jorgebucaran** [comment]
@edouard-lopez Now that https://github.com/fish-shell/fish-shell/pull/7596 has been merged, what are your goals with this plugin?

**Comment by edouard-lopez** [comment]
There is more flexibility in usage with a dedicated repo and maybe more discoverability than `fish_config`.

Personally, I prefer this as I can automate my shell configuration

**Comment by edouard-lopez** [comment]
@jorgebucaran does it work as expected for you?

**Review by jorgebucaran — APPROVED** [review]

**Pull request merged**